v3/integrations/nrfasthttp/segment.go

```diff
@@ -0,0 +1,56 @@
+package nrfasthttp
+
+import (
+	"net/http"
+
+	"github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/valyala/fasthttp"
+	"github.com/valyala/fasthttp/fasthttpadaptor"
+)
+
+func StartExternalSegment(txn *newrelic.Transaction, ctx *fasthttp.RequestCtx) *newrelic.ExternalSegment {
+	var secureAgentEvent any
+
+	if nil == txn {
+		txn = transactionFromRequestContext(ctx)
+	}
+	request := &http.Request{}
+
+	fasthttpadaptor.ConvertRequest(ctx, request, true)
+	s := &newrelic.ExternalSegment{
+		StartTime: txn.StartSegmentNow(),
+		Request:   request,
+	}
+
+	if newrelic.IsSecurityAgentPresent() {
+		secureAgentEvent = newrelic.GetSecurityAgentInterface().SendEvent("OUTBOUND", request)
+		s.SetSecureAgentEvent(secureAgentEvent)
+	}
+
+	if request != nil && request.Header != nil {
```

Comment:
Since we're moving this code, it may be a good idea to also patch the [small issues related to the security agent](https://github.com/k2io/go-agent-fork/commit/15a72c8992aa1767f7b2596ed84d07307a1d23d3).

Reply:
this should address https://new-relic.atlassian.net/browse/NR-169080 now